packages/image_picker/image_picker/example/android/app/src/test/java/io/flutter/plugins/imagepicker/ImagePickerPluginTest.java

```diff
@@ -149,6 +153,20 @@ public void onConstructor_WhenContextTypeIsActivity_ShouldNotCrash() {
         "No exception thrown when ImagePickerPlugin() ran with context instanceof Activity", true);
   }
 
+  @Test
+  public void constructDelegate_ShouldUseInternalCacheDirectory() {
+    File mockDirectory = new File("/mockpath");
+    when(mockActivity.getCacheDir()).thenReturn(mockDirectory);
+
+    ImagePickerDelegate delegate = plugin.constructDelegate(mockActivity);
+
+    verify(mockActivity, times(1)).getCacheDir();
+    assertThat(
+        "Delegate uses cache directory for storing camera captures",
+        delegate.externalFilesDirectory,
+        equalTo(mockDirectory));
```

Comment:
It would be great if we also check (in another unit test) that calling ` delegate.takeImageWithCamera`  writes a file to the mock directory.

Reply:
Fair point! A test for this has been added.